src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -105,19 +106,19 @@ export const GlobalVolatileContext = createContext<Readonly<GlobalVolatile>>({}
 export const GlobalContext = createContext<Readonly<Global>>({} as Global);
 
 const createNodeImpl = (
-    { position, data, nodeType }: NodeProto,
+    { id, position, data, nodeType }: NodeProto,
```

Comment:
I can already see use using the wrong ID variable, so let's use a default value and remove `newId`.

```suggestion
    { id = createUniqueId(), position, data, nodeType }: NodeProto,
```